### PR TITLE
Update terminology to match Cloud UI interfaces

### PR DESCRIPTION
**Pull request by meker12**
## Purpose of this pull request
Replaced deprecated UI labels with current name:

- Replace "go live" with "site launch"
- Replace "Project Owner" with "Account Owner" (Clarify License owner ==> Account owner)
- Rename Go live and launch topic to Site launch
- Rename Go live checklist to Launch checklist to match download

## Affected DevDocs pages

- https://devdocs.magento.com/cloud/architecture/pro-develop-deploy-workflow.html
- https://devdocs.magento.com/cloud/basic-information/starter-architecture.html
- https://devdocs.magento.com/cloud/basic-information/starter-develop-deploy-workflow.html
- https://devdocs.magento.com/cloud/before/before-workspace.html
- https://devdocs.magento.com/cloud/docker/docker-development.html
- https://devdocs.magento.com/cloud/env/environments-start.html
- https://devdocs.magento.com/cloud/live/go-live-checklist.html
- https://devdocs.magento.com/cloud/live/launch-steps.html
- https://devdocs.magento.com/cloud/live/live.html
- https://devdocs.magento.com/cloud/live/stage-prod-live.html
- https://devdocs.magento.com/cloud/project/projects.html
- https://devdocs.magento.com/cloud/reference/discover-deploy.html
- https://devdocs.magento.com/cloud/requirements/cloud-requirements.html
- https://devdocs.magento.com/cloud/setup/first-time-setup.md

### CODE REVIEW

File: src/cloud/basic-information/starter-develop-deploy-workflow.md [hunk 1]
```diff
@@ -203,14 +203,14 @@ To learn more, see [Deploy your store]({{ site.baseurl }}/cloud/live/stage-prod-
 
 ### Push to Master / Production {#pro}
 
-When you push to the `master` branch, you are pushing to the Production environment. Complete configuration and testing activities in the Production environment like you did in the Staging environment with one important difference. In the Production environment use live credentials for during configuration and testing. The moment you go live and launch, customers must be able to complete purchases and administrators should be able to manage your live store.
+When you push to the `master` branch, you are pushing to the Production environment. Complete configuration and testing activities in the Production environment like you did in the Staging environment with one important difference. In the Production environment use live credentials for during configuration and testing. The moment you launch your site,  customers must be able to complete purchases and administrators should be able to manage your live store.
 To learn more, see [Deploy your store]({{ site.baseurl }}/cloud/live/stage-prod-live.html).
 
-### Go live {#go-live}
+### Site launch {#go-live}
```
Comment (hguthrie) [inline src/cloud/basic-information/starter-develop-deploy-workflow.md:209]: I would love to get rid of the anchor, too. I think `site-launch` is a better one anyway.

File: src/cloud/basic-information/starter-develop-deploy-workflow.md [hunk 2]
```diff
@@ -203,14 +203,14 @@ To learn more, see [Deploy your store]({{ site.baseurl }}/cloud/live/stage-prod-
...
 
 We provide a clear walk-through for going live and launching, which requires more steps than pressing a button. After you complete these steps, your store can serve up products in your customized theme for sale immediately.
 
-To learn more, check out [Go live and launch]({{ site.baseurl }}/cloud/live/live.html).
+To learn more, see [Site launch]({{ site.baseurl }}/cloud/live/live.html).
```
Comment (hguthrie) [inline src/cloud/basic-information/starter-develop-deploy-workflow.md:213]:
```suggestion
See [Site launch]({{ site.baseurl }}/cloud/live/live.html).
```

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Pull request merged**